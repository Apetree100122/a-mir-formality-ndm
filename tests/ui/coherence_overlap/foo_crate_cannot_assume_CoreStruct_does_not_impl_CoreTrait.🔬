[
    crate core {
        trait CoreTrait<> where [] {}
        struct CoreStruct<> where [] {}
    },
    crate foo {
        trait FooTrait<> where [] {}
        impl<ty T> FooTrait<> for T where [T: CoreTrait<>] {}
        impl<> FooTrait<> for CoreStruct<> where [] {}
    }
]
