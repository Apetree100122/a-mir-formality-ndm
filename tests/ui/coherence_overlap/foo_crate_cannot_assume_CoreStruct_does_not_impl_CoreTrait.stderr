Error: impls may overlap:
impl <ty> FooTrait < > for ^ty0_0 where [^ty0_0 : CoreTrait < >] { }
impl <> FooTrait < > for (rigid (adt CoreStruct)) where [] { }
